mmv1/third_party/terraform/resources/resource_compute_security_policy.go.erb

```diff
@@ -150,6 +150,56 @@ func resourceComputeSecurityPolicy() *schema.Resource {
 							Description: `A match condition that incoming traffic is evaluated against. If it evaluates to true, the corresponding action is enforced.`,
 						},
 
+						<% unless version == 'ga' -%>
+						"preconfigured_waf_config": {
```

Comment:
We usually want to prevent users from specify empty blocks, so I would recommend that we make the only subfield `exclusion` required. However, it is not a requirement, as I see you set an empty block `preconfigured_waf_config{}` in the test and assume that empty blocks may be needed for some purposes in this case?

Reply:
That is correct, the empty block is required here in order to clear what's inside it from the API object as well as avoid perma-drift. Originally I had `exclusion` marked as required, but changed it to optional so that the provider will cover the edge cases here. And that's why the test is there with the empty block, because I ran into this issue myself and so added the test when setting `exclusion` as optional to ensure it's possible to remove the exclusions from the rule properly.

It's a bit weird, but it's due the `preconfiguredWafConfig` field not existing on the API object until you first set it, but once included the only way to remove the exclusions it contains is to actually send an empty object, after-which `preconfiguredWafConfig: {}` is what will come back in the YAML output from describe.